modules/programs/micro/default.nix

```diff
@@ -0,0 +1,41 @@
+{ config, lib, pkgs, ... }:
+let
+  cfg = config.programs.micro;
+  dir = ".config/micro";
+in {
+  meta.maintainers = [ maintainers.sivizius ];
+
+  ###### interface
+  options = {
+    programs.micro = {
+      enable = lib.mkEnableOption
+        "micro – a modern easy-to-use intuitive terminal-based text editor.";
+      bindings = import ./bindings.nix lib;
+      colorschemes = import ./colorschemes.nix lib;
+      extraSettings = import ./extraSettings.nix lib;
+      plugins = import ./plugins.nix lib;
+      settings = import ./settings.nix lib;
```

Comment:
Change to a single option having a suitable type for a RFC-42 style settings option. Something like
```nix
with types; let
  primitive = oneOf [ str bool int (listOf str) ];
  localSettings = attrsOf primitive;
in attrsOf (either primitive localSettings) // { description = "Micro configuration"; }
```
might suffice.